web/src/auth/SignupPage.js

```diff
@@ -499,6 +500,40 @@ class SignupPage extends React.Component {
     )
   }
 
+  renderProviderLogo(provider, application, width, margin, size) {
```

Comment:
Don't duplicate the code, put the shared code into Setting.js

Reply:
In fact, the redner function here does not need to use size, and the logic of processing saml is not clear now.

Reply:
@qianxi0410 that's another issue. If you think it's an issue and want to fix it, you need to make a new PR